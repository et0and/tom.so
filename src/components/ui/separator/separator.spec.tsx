import React from "react";
import { render } from "@testing-library/react";
import { Separator } from "./separator";

describe("<Separator/>", () => {
  it("renders correctly with default props", () => {
    const { container } = render(<Separator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with vertical orientation", () => {
    const { container } = render(<Separator orientation="vertical" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with custom className", () => {
    const { container } = render(<Separator className="custom-class" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with decorative set to false", () => {
    const { container } = render(<Separator decorative={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
