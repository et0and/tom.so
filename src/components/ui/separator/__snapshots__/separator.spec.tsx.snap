// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator/> renders correctly with custom className 1`] = `
<div
  class="shrink-0 bg-gray-200 dark:bg-gray-400 h-[1px] w-full custom-class"
  data-orientation="horizontal"
  role="none"
/>
`;

exports[`<Separator/> renders correctly with decorative set to false 1`] = `
<div
  class="shrink-0 bg-gray-200 dark:bg-gray-400 h-[1px] w-full"
  data-orientation="horizontal"
  role="separator"
/>
`;

exports[`<Separator/> renders correctly with default props 1`] = `
<div
  class="shrink-0 bg-gray-200 dark:bg-gray-400 h-[1px] w-full"
  data-orientation="horizontal"
  role="none"
/>
`;

exports[`<Separator/> renders correctly with vertical orientation 1`] = `
<div
  class="shrink-0 bg-gray-200 dark:bg-gray-400 h-full w-[1px]"
  data-orientation="vertical"
  role="none"
/>
`;
